### PR TITLE
WIFI-14152: ucentral-schema: add support for configuring snmpd service using ucentral 

### PR DESCRIPTION
**Pull request by fjamarquez**
These changes make it possible to configure the snmpd service from the json file.

Example:
```   "services": {
        "snmpd": {
	    "agent": {
		"agentaddress": "UDP:161"
	    },
            "access": {
		"private_access": {
                       "context": "none",
                       "group": private",
                       "level": "noauth",
                       "notify": "none",
                       "prefix": "exact",
                       "read": "all",
                       "version": "any",
                       "write": "none"
                },
		"public_access": {
                       "context": "none",                                                                
                       "group": "public",                                                                 
                       "level": "noauth",                                                                
                       "notify": "none",                                                                 
                       "prefix": "exact",                                                                
                       "read": "all",                                                                    
                       "version": "any",                                                                 
                       "write": "none"                                                                   
                } 
	    },
            "agentx": {
                "type": "master"
            },
            "com2sec": {
                "public": {
                        "community": "public",
                        "secname": "ro",
                        "source": "default"
                 },
                 "private": {
                        "community": "private",
                        "secname": "rw",
                        "source": "default"
                 }
            },
            "general": {
                "enabled": true,
            },
            "pass": [
                {                               
                        "miboid": ".1.3.6.1.1",
                        "name": "pass protocl",
                        "prog": "/bin/ash /etc/snmp/script.sh"
                },                       
            ], 
            "group": {
                "private_v2c": {
                        "group": "private",
                        "secname": "rw",
                        "version": "v2c"
                },
                "public_v2c": {
                        "group": "public",
                        "secname": "ro",
                        "version": "v2c"
                 }
            },
            "system": {
                "sysContact": "company.com",
                "sysLocation": "Spain",
                "sysName": "Host1"
            },
            "view": {
                "all": {
                        "oid": ".1",
                        "type": "included",
                        "viewname": "all"
                 }
            }
        },`

### CODE REVIEW

File: schema/service.snmpd.access.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+description:
+  List of access types for SNMP.
+type: object
+properties:
+  public_access:
```
Comment (blogic) [inline schema/service.snmpd.access.yml:5]: could you change the "_" to "-" throughout the schema please ? the schemareader will auto convert the config to use "_"

Comment (fjamarquez) [reply schema/service.snmpd.access.yml]: yes, of course. I have made the changes. If you see any other issue, please let me know. I'm not familiar with syntaxt yet, sorry.

**Comment by fjamarquez** [comment]
> @fjamarquez looks good. How big is the snmp binary ? shall we always include it in the default images ?

we are not obliged to always compile the snmpd binary. If the binary is not compiled, the snmpd configuration will not beapplied. If we want to use snmpd always, then we should compiled it. I think that depends on the case.
The snmpd binary size is 64.0K. There are anothers snmpd packages available as snmpd-mibs or snmpd-utils.
I can create a new profile with snmpd package like that:

```
---
description: Add the SNMP packages
packages:
  - snmp-mibs
  - snmp-utils
  - snmpd
  - libnetsnmp
```

We can include it whenever we want.

**Comment by blogic** [comment]
@fjamarquez I will bring this up in the thursday call and see what community thinks about default enabling snmp

**Comment by fjamarquez** [comment]
> @fjamarquez I will bring this up in the thursday call and see what community thinks about default enabling snmp

ok, perfect! it is a service that does nothing to improve wifi performance, it only serves to monitor network devices such as aps. The request must be made directly to each ap. It is another way to monitor the devices in addition to the cloud.  It is mainly used by network administrators to obtain device information.

**Comment by blogic** [comment]
merged, Thanks !

**Comment by fjamarquez** [comment]
Hi @blogic , we have detected a small typo in the parser in our last test, i see that this has been postponed  for the release 4, so could you reopen the MR to update with the fork branch and fix it? Sorry for that.

**Pull request closed without merging**